### PR TITLE
Debugger and Source tabs for the Core Build launch configuration.

### PR DESCRIPTION
**Pull request by ewaterlander**
The launch configuration for local Core Build projects got the Debugger and Source tabs in debug mode. Now the user can set the stop at startup, a specific debugger, and all other Debug options as we know from Managed Build projects.

The GdbLaunch class automatically picks up all the debug settings from these new tabs. No additional code was needed for that.

The composition of Core Build tab groups is now fully done via the launchConfigurationTabGroups extension-point.

This change relates to #758. It affects all Core Build projects, including CMake projects.

### CODE REVIEW

**Comment by ewaterlander** [comment]
image

**Comment by ewaterlander** [comment]
Hi @betamaxbandit and @jonahgraham ,
would you like to review this change?
regards,
Erwin

**Comment by ewaterlander** [comment]
Hi,
I had to update the major version of org.eclipse.cdt.launch. I set it to 11.0.000 and now the check reports this error:

Error:  Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:4.0.6:validate-version (default-validate-version) on project org.eclipse.cdt.launch: Unqualified OSGi version 11.0.0.qualifier must match unqualified Maven version 11.0.000-SNAPSHOT for SNAPSHOT builds -> [Help 1]

I don't know where to fix this.

**Comment by ewaterlander** [comment]
Hi @jonahgraham 
Can you help me with the version numbers?

**Comment by betamaxbandit** [comment]
> Hi @betamaxbandit and @jonahgraham , would you like to review this change? regards, Erwin

Hi @ewaterlander ,
Thanks for the patch. I'm a little busy with other things right now, but I'll try and have a look in the next few days.
Cheers John

**Review by jonahgraham — APPROVED** [review]
> Can you help me with the version numbers?

Yes - it turns out that `11.0.000` in invalid, this was new to me. I added a commit which is validating now that changes it to `11.0.0` and it passed for me locally.

I'll squash the change (and the minor API `@since` change too) when I merge.

@betamaxbandit I am fine with this change, LMK if you have concerns at a high level or want more time before I merge.

---

FYI @kolipakakondal - as one of the known consumers of core build I wanted to make sure you saw this change that is upcoming in CDT 12.

File: launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/corebuild/LocalLaunchConfigurationTabGroup.java [hunk 1]
```diff
@@ -18,6 +18,13 @@
 import org.eclipse.debug.ui.ILaunchConfigurationDialog;
 import org.eclipse.debug.ui.ILaunchConfigurationTab;
 
+/**
+ * Tab group for local Core Build projects.
+ *
+ * @deprecated in 11.0, because the composition of the Core Build local tab group is defined
+ * via the launchConfigurationTabGroups extension-point.
+ */
+@Deprecated
```
[the remark this answers is not in the document]
Comment (ewaterlander) [reply launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/corebuild/LocalLaunchConfigurationTabGroup.java:27]: I'm fine with deleting.

[the remark this answers is not in the document]
Comment (jonahgraham) [reply launch/org.eclipse.cdt.launch/src/org/eclipse/cdt/launch/internal/corebuild/LocalLaunchConfigurationTabGroup.java]: I pushed a commit with the deletion.

**Comment by jonahgraham** [comment]
There are test failures - all known in either https://github.com/eclipse-cdt/cdt/issues/816 or https://github.com/eclipse-cdt/cdt/issues/194 - besides that the CI run is good and this is ready for merge.

**Review by betamaxbandit — APPROVED** [review]
LGTM

**Comment by kolipakakondal** [comment]
Thanks @jonahgraham for letting us know

**Comment by kolipakakondal** [comment]
Hi @jonahgraham How do I take the update site build with these changes?

**Comment by jonahgraham** [comment]
@kolipakakondal Please see [download development builds](https://github.com/eclipse-cdt/cdt/blob/main/Downloads.md#development-builds) info for how to get bleeding edge builds that will have this change from build \#454. When CDT 12.0 M1 is published in the new year you can pick it up as a development build.

**Pull request merged**